src/Sphere.jl

```diff
@@ -156,3 +156,75 @@ function normal_tvector_distribution(S::Sphere, x, σ)
     d = Distributions.MvNormal(zero(x), σ)
     return ProjectedFVectorDistribution(TangentBundleFibers(S), x, d, project_vector!, x)
 end
+
+struct GeodesicInterpolationMethod <: AbstractMethod end
+
+"""
+    mean!(
+        S::Sphere,
+        y,
+        x::AbstractVector,
+        w::AbstractWeights,
+        method::GeodesicInterpolationMethod;
+        shuffle_rng=nothing,
+        kwargs...,
+    )
+
+Compute the Riemannian mean (center of mass) of `x`. The algorithm is adapted
+from the below paper to handle the weighted mean. It uses repeated weighted
+geodesic interpolation and converges to the Riemannian mean when all `x` are
+within a distance of π/2 of the mean.
+
+If `shuffle_rng` is provided, it is used to shuffle the order in which the
+points are considered for computing the mean.
+
+For more information on the geodesic interpolation method, see the below paper:
+> Salehian H et al: An efficient recursive estimator of the
+> Fréchet mean on a hypersphere with applications to Medical Image Analysis,
+> Mathematical Foundations of Computational Anatomy (2015).
```

Comment:
Could you add a doi? Do they have a freely available version (arXiv/HAL) we could add?

Reply:
Unfortunately, there's no DOI (that I can find). It looks like this was a pretty small conference (led by Xavier Pennec). The official posting of the paper is here:https://www-sop.inria.fr/asclepios/events/MFCA15/Papers/MFCA15_4_2.pdf, but I don't know how stable the site/URL is. If we have link checking turned on, it should be fine. Think I should just hyperlink the title to it?

Reply:
I think it was published as a book chapter, but you're right - that one is hard to track down.